### PR TITLE
Add SearchMarcros containing a copy of DTK_MARK_REGION

### PR DESCRIPTION
**Pull request by aprokop**
This breaks the last dependency of Search on DTK proper.

### CODE REVIEW

File: packages/Search/src/DTK_SearchMacros.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DTK_SEARCH_MACROS_HPP
+#define DTK_SEARCH_MACROS_HPP
+
+#define DTK_MARK_REGION( x ) std::string( "DTK_" ) + x
+
+#endif
```
Comment (dalg24) [inline packages/Search/src/DTK_SearchMacros.hpp]: I'd rather have it in `details/` for now

**Comment by aprokop** [comment]
> What happen if downstream code ends up transitively including both headers `<DTK_ConfigDefs.hpp>` and `<DTK_SearchMacros.hpp>`?

They'll get a warning about redefining macros. I can replace the name, but `DTK_SEARCH_MARK_REGION` sounded too cumbersome to me.

File: packages/Search/src/DTK_SearchMacros.hpp (hunk 1 above)
Comment (aprokop) [reply packages/Search/src/DTK_SearchMacros.hpp]: Could do that.

**Comment by dalg24** [comment]
how about if not def?

File: packages/Search/src/details/DTK_SearchMacros.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DTK_SEARCH_MACROS_HPP
+#define DTK_SEARCH_MACROS_HPP
+
+#define DTK_SEARCH_MARK_REGION( x ) std::string( "DTK_Search_" ) + x
```
Comment (sslattery) [inline packages/Search/src/details/DTK_SearchMacros.hpp]: What is the upside of even having this macro now that we have reduced what it does to adding a simple prefix?

Comment (aprokop) [reply packages/Search/src/details/DTK_SearchMacros.hpp]: So that you don't need to add this prefix manually each time and be consistent throughout.

**Comment by dalg24** [comment]
retest this please

**Comment by aprokop** [comment]
> retest this please

Does this redo Jenkins job?

**Comment by aprokop** [comment]
> ```c++
> #if !defined( DTK_MARK_REGION )
>   #define DTK_MARK_REGION ...
> #endif 
> ```

Could do that, but you already merged it....

**Pull request merged**